code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -89,6 +89,13 @@
 	C.AdjustImmobilized(-10 * removed)
 	. = TRUE
 
+/datum/reagent/drug/nicotine/on_mob_metabolize(mob/living/carbon/C, class)
+	. = ..()
+	C.stats?.set_skill_modifier(1, /datum/rpg_skill/handicraft, SKILL_SOURCE_NICOTINE)
```

Comment:
See other uses of on_mob_metabolize(). The class argument is passed which indicates what type of metabolism is happening.